src/Tasks/Microsoft.NET.Build.Tasks/NuGetUtils.NuGet.cs

```diff
@@ -82,20 +82,38 @@ bool FileMatchesProjectLanguage()
 
         public static string GetBestMatchingRid(RuntimeGraph runtimeGraph, string runtimeIdentifier,
             IEnumerable<string> availableRuntimeIdentifiers, out bool wasInGraph)
+        {
+            return GetBestMatchingRidWithExclusion(runtimeGraph, runtimeIdentifier,
+                runtimeIdentifiersToExclude: null,
+                availableRuntimeIdentifiers, out wasInGraph);
+        }
+
+        public static string GetBestMatchingRidWithExclusion(RuntimeGraph runtimeGraph, string runtimeIdentifier,
+            IEnumerable<string> runtimeIdentifiersToExclude,
+            IEnumerable<string> availableRuntimeIdentifiers, out bool wasInGraph)
         {
             wasInGraph = runtimeGraph.Runtimes.ContainsKey(runtimeIdentifier);
 
+            string bestMatch = null;
+
             HashSet<string> availableRids = new HashSet<string>(availableRuntimeIdentifiers);
+            HashSet<string> excludedRids = runtimeIdentifiersToExclude switch { null => null, _ => new HashSet<string>(runtimeIdentifiersToExclude) };
```

Comment:
Is it worth using the ctor for `HashSet<T>` where you pass a `StringComparer`? `StringComparer.Ordinal` might help performance a bit? (`Or OrdinalIgnoreCase` if it is correct)